clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2848,8 +2848,12 @@ class SYCLKernelNameTypeVisitor
     if (T.isNull())
       return;
     const CXXRecordDecl *RD = T->getAsCXXRecordDecl();
-    if (!RD)
+    if (!RD) {
+      if (T->isNullPtrType())
+        S.Diag(KernelInvocationFuncLoc, diag::err_sycl_kernel_incorrectly_named)
+            << /* kernel name cannot be a type in the std namespace */ 3;
```

Comment:
The original [code](https://github.com/intel/llvm/blob/sycl/clang/lib/Sema/SemaSYCL.cpp#L3445) did not set the kernel invocation function to be invalid if the kernel name happens to be a nullptr type. Hence, I reverted the "IsInvalid = true" change. If the kernel function indeed has to be made invalid for this case, I will set "IsInvalid" to true, but it causes a bunch of failures in both FE and runtime which needs to be fixed.

Reply:
I would assume this should be set to invalid. Can you elaborate on failures you see?

Reply:
The failures were because the initial patch did not handle kernel name types with template parameter pack.
Ex:
```
namespace std { typedef decltype(nullptr) nullptr_ns; }
typedef decltype(nullptr) nullptr_t;
template <typename T, typename... Args> class TemplParamPack;

cgh.single_task<TemplParamPack<int, float, std::nullptr_ns, double>>([]() {});
cgh.single_task<TemplParamPack<int, float, nullptr_t, double>>([]() {});

```